### PR TITLE
small cleanup

### PR DESCRIPTION
**Pull request by MauricioUyaguari**
## Summary

Addresses comments on: https://github.com/finos/legend-studio/pull/1114

## How did you test this change?

- [ ] Test(s) added
- [ ] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

File: packages/legend-graph/src/graph/PureGraphPlugin.ts [hunk 1]
```diff
@@ -50,5 +50,8 @@ export abstract class PureGraphPlugin extends AbstractPlugin {
    */
   getExtraDeadReferencesCleaners?(): DeadReferencesCleaner[];
 
-  getExtraTestables?(): TestableExtension[];
+  /**
+   * Get the list of extra collectors that return items that implement the `testable` interface
```
Comment (akphi) [inline packages/legend-graph/src/graph/PureGraphPlugin.ts:54]: --> `Get the list of collectors for testable items

File: packages/legend-graph/src/graphManager/PureGraphManagerPlugin.ts [hunk 2]
```diff
@@ -73,4 +77,8 @@ export abstract class PureGraphManagerPlugin extends AbstractPlugin {
    * Get the list of Pure grammar element labelers.
    */
   getExtraPureGrammarElementLabelers?(): PureGrammarElementLabeler[];
+
+  getExtraIdFromTestableGetters?(): IdFromTestableGetter[];
```
Comment (akphi) [inline packages/legend-graph/src/graphManager/PureGraphManagerPlugin.ts:81]: --> `TestableIDBuilder`

File: packages/legend-graph/src/graphManager/AbstractPureGraphManager.ts [hunk 3]
```diff
@@ -219,6 +220,7 @@ export abstract class AbstractPureGraphManager {
   // ------------------------------------------- Test  -------------------------------------------
   abstract runTests(
```
Comment (akphi) [inline packages/legend-graph/src/graphManager/AbstractPureGraphManager.ts:221]: let's a new line here

File: packages/legend-graph/src/models/metamodels/pure/test/Test.ts [hunk 4]
```diff
@@ -24,7 +24,7 @@ export abstract class Test implements Hashable {
 }
 
 export abstract class AtomicTest extends Test implements Hashable {
-  parentSuite: TestSuite | undefined;
+  __parentSuite: TestSuite | undefined;
```
[the remark this answers is not in the document]
Comment (akphi) [reply packages/legend-graph/src/models/metamodels/pure/test/Test.ts:27]: or `reaodnly _PARENT`

File: packages/legend-graph/src/models/protocols/pure/v1/engine/test/V1_RunTestsResult.ts [hunk 5]
```diff
@@ -33,8 +38,13 @@ export class V1_RunTestsResult {
 
 export const V1_buildTestsResult = (
   results: V1_RunTestsResult,
-  graph: PureModel,
+  testableFromIdGetter: (id: string) => Testable | undefined,
```
Comment (akphi) [inline packages/legend-graph/src/models/protocols/pure/v1/engine/test/V1_RunTestsResult.ts:41]: --> `testableFinder`

File: packages/legend-studio/src/stores/sidebar-state/testable/GlobalTestRunnerState.ts [hunk 6]
```diff
@@ -242,8 +296,19 @@ const getTestableResultFromTestResults = (
 
 export const getNodeTestableResult = (
   node: TestableExplorerTreeNodeData,
+  gloablRun: boolean,
   results: Map<AtomicTest, TestResult>,
 ): TESTABLE_RESULT => {
+  if (gloablRun && node instanceof TestableTreeNodeData) {
```
Comment (akphi) [inline packages/legend-studio/src/stores/sidebar-state/testable/GlobalTestRunnerState.ts:302]: typo in `gloablRun`

File: packages/legend-studio/src/stores/sidebar-state/testable/GlobalTestRunnerState.ts [hunk 7]
```diff
@@ -295,50 +360,71 @@ export class TestableState {
...
+      editorStore,
+      this.globalTestRunnerState.extraTestableMetadataGetters,
     );
     this.treeData = this.buildTreeData(this.testableMetadata);
   }
 
   *run(node: TestableExplorerTreeNodeData): GeneratorFn<void> {
+    this.isRunningTests.inProgress();
+    let input: RunTestsTestableInput;
+    let nodeRunning = node;
```
Comment (akphi) [inline packages/legend-studio/src/stores/sidebar-state/testable/GlobalTestRunnerState.ts:375]: --> `currentNode`

File: packages/legend-graph/src/graphManager/PureGraphManagerPlugin.ts (hunk 2 above)
Comment (akphi) [reply packages/legend-graph/src/graphManager/PureGraphManagerPlugin.ts:81]: Also, let's add doc for these 2

Comment (akphi) [reply packages/legend-graph/src/graphManager/PureGraphManagerPlugin.ts:81]: These should also go to a separate `DSLData_PureGraphManagerPlugin_Extension`

Comment (MauricioUyaguari) [reply packages/legend-graph/src/graphManager/PureGraphManagerPlugin.ts:81]: no they should not go to DSLData. They are not part of DSLDATA

Comment (MauricioUyaguari) [reply packages/legend-graph/src/graphManager/PureGraphManagerPlugin.ts]: Test and Data are two different concepts and Testable should not belong under the DSL_Data

File: packages/legend-graph/src/graph/PureGraphPlugin.ts [hunk 8]
```diff
@@ -50,5 +50,8 @@ export abstract class PureGraphPlugin extends AbstractPlugin {
    */
   getExtraDeadReferencesCleaners?(): DeadReferencesCleaner[];
 
-  getExtraTestables?(): TestableExtension[];
+  /**
+   * Get the list of collectors for testable items
+   */
+  getExtraTestableCollectors?(): TestablesGetter[];
```
Comment (akphi) [inline packages/legend-graph/src/graph/PureGraphPlugin.ts:56]: --> `getExtraTestablesCollectors?(): TestablesCollector[]`

File: packages/legend-graph/src/graphManager/AbstractPureGraphManager.ts [hunk 9]
```diff
@@ -217,8 +218,10 @@ export abstract class AbstractPureGraphManager {
   ): Promise<string>;
 
   // ------------------------------------------- Test  -------------------------------------------
+
   abstract runTests(
     graph: PureModel,
+    graphManagerState: GraphManagerState,
```
Comment (akphi) [inline packages/legend-graph/src/graphManager/AbstractPureGraphManager.ts:224]: No, this doesn't look right. We must not pass any `state/store` here, wrong layer

File: packages/legend-graph/src/models/metamodels/pure/test/Testable.ts [hunk 10]
```diff
@@ -16,31 +16,49 @@
...
+export type TestableFinder = (
   id: string,
   graph: PureModel,
+) => Testable | undefined;
+
+export type TestablesGetter = (graph: PureModel) => Testable[];
+
+export const getNullableTestable = (
+  id: string,
+  graphManager: GraphManagerState,
```
Comment (akphi) [inline packages/legend-graph/src/models/metamodels/pure/test/Testable.ts:41]: we can't pass a state here, so instead, let's pass the `graph` and the list of `PureGraphManager` plugins.

Also, we should not have logic in `metamodel` so let's move this out to `src/helper/Testable_Helper.ts`

File: packages/legend-graph/src/graphManager/AbstractPureGraphManager.ts (hunk 9 above)
Comment (akphi) [reply packages/legend-graph/src/graphManager/AbstractPureGraphManager.ts:224]: See my note in `Testable.ts` so we can pass in more appropriate parameters here

File: packages/legend-graph/src/graphManager/PureGraphManagerPlugin.ts (hunk 2 above)
Comment (akphi) [reply packages/legend-graph/src/graphManager/PureGraphManagerPlugin.ts]: yep, my bad

**Review by akphi — CHANGES_REQUESTED** [review]

File: packages/legend-graph/src/graphManager/AbstractPureGraphManager.ts [hunk 11]
```diff
@@ -217,8 +218,10 @@ export abstract class AbstractPureGraphManager {
   ): Promise<string>;
 
   // ------------------------------------------- Test  -------------------------------------------
+
   abstract runTests(
     graph: PureModel,
+    pureGraphManagerPlugins: PureGraphManagerPlugin[],
```
Comment (akphi) [inline packages/legend-graph/src/graphManager/AbstractPureGraphManager.ts:224]: Just remember that `AbstractPureGraphManager` has access to these plugins anyway, we should also swap order of `graph` and `testableInputs`

File: packages/legend-graph/src/models/metamodels/pure/test/Testable.ts [hunk 12]
```diff
@@ -16,31 +16,49 @@
...
+) => string | undefined;
+
+export type TestableFinder = (
+  id: string,
+  graph: PureModel,
+) => Testable | undefined;
+
+export type TestablesCollector = (graph: PureModel) => Testable[];
+
 export const getNullableTestable = (
```
Comment (akphi) [inline packages/legend-graph/src/models/metamodels/pure/test/Testable.ts:39]: Let's move these to helper

**Review by akphi — APPROVED** [review]

**Pull request merged**